### DISCUSSION TITLE
support pyproject.toml as well as requirements.txt

### DISCUSSION THREAD

**Discussion by Imaclean74** [brettcannon/WWBD#30] [Ideas]
ie - poetry support

**Comment by brettcannon** [comment]
This is a bit tricky as what tool would we use to do the installation (e.g. how do we know it's poetry, or Hatch, etc.)? And what about extras specified in `pyproject.toml`?

**Comment by Imaclean74** [comment]
is this block in pyproject.toml ?

```
[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```
not sufficient ?

**Reply by brettcannon** [reply]
No because all that's for is installing the build tool for your project. Since WWBD has nothing to do with building a project, it isn't relevant. What I was referring to is the [`project.dependencies` entry](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#dependencies-optional-dependencies).

What would have to be decided is:

1. How to handle optional dependencies?
2. What about editable installs (which is how you typically install these sorts of dependencies)?
3. What to use to perform the installations?

Do note I am the co-author of `pyproject.toml` and the standards surrounding it, so I would be a bit surprised if there wasn't _some_ support someday. 😉 But I wanted to start with simpler, easier-to-design-and-explain approach with requirements files as e.g. students are much more likely to be using them than to be creating a library that entails `pyproject.toml`.

**Reply by mattmess1221** [reply]
Usually, the presence of a `poetry.lock` file means the project uses poetry.

**Comment by Imaclean74** [comment]
makes sense. Thanks for the explanation. However - at least for my usage, I tend to use 'poetry new' even for small throwaway scripting projects, because it's clear and simple.  I may not be the typical target user of WWBD though.

**Reply by brettcannon** [reply]
If we were to support `pyproject.toml`, it would be via `pip install -e .` and not via specific tools like Poetry. Basically the flow is for people who are so new they don't know what Poetry is. 😁 We are starting to talk about letting other extensions participate in some sort of environment creation/management flow, which would let Poetry be used appropriately.

**Comment by parched** [comment]
This reminds me of https://github.com/microsoft/Oryx/issues/1412, given that the vscode-python surveys seem to promote using Azure, it might be worth syncing up.

**Reply by brettcannon** [reply]
Thanks for the pointer! I have left a comment over there.
